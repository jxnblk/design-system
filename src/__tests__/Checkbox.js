import React from 'react'
import renderer from 'react-test-renderer'
import { Checkbox, Icon, theme } from '..'
import { shallow, mount, render } from 'enzyme'

let onChangeFn = jest.fn()
const name = 'random_check_box'
const checkBoxSelector = `#${name}`

describe('Checkbox', () => {
  beforeEach(() => {
    onChangeFn.mockReset()
  })

  test('renders without the theme passed specifically', () => {
    const component = <Checkbox id={name} onChange={onChangeFn} />
    const wrapper = mount(component)
    const json = renderer.create(component).toJSON()

    expect(json).toMatchSnapshot()
  })

  test('renders with the theme passed specifically', () => {
    const component = <Checkbox id={name} onChange={onChangeFn} theme={theme} />
    const wrapper = shallow(component)
    const json = renderer.create(component).toJSON()

    expect(json).toMatchSnapshot()
  })

  test('renders checked when defaultChecked prop is passed as true', () => {
    const json = renderer
      .create(<Checkbox id={name} defaultChecked onChange={onChangeFn} />)
      .toJSON()
    const [input] = json.children

    expect(input.props.defaultChecked).toBe(true)
  })

  test('renders disabled with disabled prop', () => {
    const json = renderer
      .create(<Checkbox id={name} disabled={true} onChange={onChangeFn} />)
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders disabled with defaultChecked', () => {
    const json = renderer
      .create(
        <Checkbox
          id={name}
          disabled={true}
          defaultChecked={true}
          onChange={onChangeFn}
        />
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  /** Not Testing wrapper.simulate('click) because enzyme won't pass it through to the checkbox
   * We have to test the actual element receiving the event and the label 'click' causes a 'change'
   * for the nested input. This also means we cannot accurately test that disabled checkboxes ignore clicks.
   */
  test('updates and fires callback when clicked (when enabled)', () => {
    const component = <Checkbox id={name} onChange={onChangeFn} />
    const wrapper = shallow(component)
    const json = renderer.create(component).toJSON()
    wrapper.find('[type="checkbox"]').simulate('change', {
      target: {
        checked: true
      }
    })
    expect(onChangeFn).toHaveBeenCalled()
  })
})
