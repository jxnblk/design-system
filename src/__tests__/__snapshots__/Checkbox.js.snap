// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders checked when defaultChecked prop is passed as true 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: #687B8E;
}

.c0 svg[data-name='checked'] {
  display: none;
}

.c0 > input:checked ~ svg[data-name='checked'] {
  display: inline-block;
  color: #007aff;
}

.c0 > input:checked ~ svg[data-name='empty'] {
  display: none;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
}

<div
  className="c0"
  disabled={false}
>
  <input
    className="c1"
    defaultChecked={true}
    disabled={false}
    id="random_check_box"
    onChange={[MockFunction]}
    size={20}
    type="checkbox"
  />
  <svg
    className="c2"
    data-name="checked"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6,3h12c1.7,0,3,1.3,3,3v12c0,1.7-1.3,3-3,3H6c-1.7,0-3-1.3-3-3V6C3,4.3,4.3,3,6,3z M10,17
	l9-8.6L17.6,7L10,14.3l-3.6-3.5L5,12.2L10,17z"
    />
  </svg>
  <svg
    className="c2"
    data-name="empty"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6,5C5.4,5,5,5.4,5,6v12c0,0.6,0.4,1,1,1h12c0.6,0,1-0.4,1-1V6c0-0.6-0.4-1-1-1H6z M6,3h12
	c1.7,0,3,1.3,3,3v12c0,1.7-1.3,3-3,3H6c-1.7,0-3-1.3-3-3V6C3,4.3,4.3,3,6,3z"
    />
  </svg>
</div>
`;

exports[`Checkbox renders disabled when disabled props is passed as true 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: #d1d6db;
}

.c0 svg[data-name='checked'] {
  display: none;
}

.c0 > input:checked ~ svg[data-name='checked'] {
  display: inline-block;
  color: #d1d6db;
}

.c0 > input:checked ~ svg[data-name='empty'] {
  display: none;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
}

<div
  className="c0"
  disabled={true}
>
  <input
    className="c1"
    defaultChecked={false}
    disabled={true}
    id="random_check_box"
    onChange={[MockFunction]}
    size={20}
    type="checkbox"
  />
  <svg
    className="c2"
    data-name="checked"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6,3h12c1.7,0,3,1.3,3,3v12c0,1.7-1.3,3-3,3H6c-1.7,0-3-1.3-3-3V6C3,4.3,4.3,3,6,3z M10,17
	l9-8.6L17.6,7L10,14.3l-3.6-3.5L5,12.2L10,17z"
    />
  </svg>
  <svg
    className="c2"
    data-name="empty"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6,5C5.4,5,5,5.4,5,6v12c0,0.6,0.4,1,1,1h12c0.6,0,1-0.4,1-1V6c0-0.6-0.4-1-1-1H6z M6,3h12
	c1.7,0,3,1.3,3,3v12c0,1.7-1.3,3-3,3H6c-1.7,0-3-1.3-3-3V6C3,4.3,4.3,3,6,3z"
    />
  </svg>
</div>
`;

exports[`Checkbox renders disabled when disabled props is passed as true 2`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: #d1d6db;
}

.c0 svg[data-name='checked'] {
  display: none;
}

.c0 > input:checked ~ svg[data-name='checked'] {
  display: inline-block;
  color: #d1d6db;
}

.c0 > input:checked ~ svg[data-name='empty'] {
  display: none;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
}

<div
  className="c0"
  disabled={true}
>
  <input
    className="c1"
    defaultChecked={true}
    disabled={true}
    id="random_check_box"
    onChange={[MockFunction]}
    size={20}
    type="checkbox"
  />
  <svg
    className="c2"
    data-name="checked"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6,3h12c1.7,0,3,1.3,3,3v12c0,1.7-1.3,3-3,3H6c-1.7,0-3-1.3-3-3V6C3,4.3,4.3,3,6,3z M10,17
	l9-8.6L17.6,7L10,14.3l-3.6-3.5L5,12.2L10,17z"
    />
  </svg>
  <svg
    className="c2"
    data-name="empty"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6,5C5.4,5,5,5.4,5,6v12c0,0.6,0.4,1,1,1h12c0.6,0,1-0.4,1-1V6c0-0.6-0.4-1-1-1H6z M6,3h12
	c1.7,0,3,1.3,3,3v12c0,1.7-1.3,3-3,3H6c-1.7,0-3-1.3-3-3V6C3,4.3,4.3,3,6,3z"
    />
  </svg>
</div>
`;

exports[`Checkbox renders with the theme passed specifically 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: #687B8E;
}

.c0 svg[data-name='checked'] {
  display: none;
}

.c0 > input:checked ~ svg[data-name='checked'] {
  display: inline-block;
  color: #007aff;
}

.c0 > input:checked ~ svg[data-name='empty'] {
  display: none;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
}

<div
  className="c0"
  disabled={false}
>
  <input
    className="c1"
    defaultChecked={false}
    disabled={false}
    id="random_check_box"
    onChange={[MockFunction]}
    size={20}
    type="checkbox"
  />
  <svg
    className="c2"
    data-name="checked"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6,3h12c1.7,0,3,1.3,3,3v12c0,1.7-1.3,3-3,3H6c-1.7,0-3-1.3-3-3V6C3,4.3,4.3,3,6,3z M10,17
	l9-8.6L17.6,7L10,14.3l-3.6-3.5L5,12.2L10,17z"
    />
  </svg>
  <svg
    className="c2"
    data-name="empty"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6,5C5.4,5,5,5.4,5,6v12c0,0.6,0.4,1,1,1h12c0.6,0,1-0.4,1-1V6c0-0.6-0.4-1-1-1H6z M6,3h12
	c1.7,0,3,1.3,3,3v12c0,1.7-1.3,3-3,3H6c-1.7,0-3-1.3-3-3V6C3,4.3,4.3,3,6,3z"
    />
  </svg>
</div>
`;

exports[`Checkbox renders without the theme passed specifically 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: #687B8E;
}

.c0 svg[data-name='checked'] {
  display: none;
}

.c0 > input:checked ~ svg[data-name='checked'] {
  display: inline-block;
  color: #007aff;
}

.c0 > input:checked ~ svg[data-name='empty'] {
  display: none;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
}

<div
  className="c0"
  disabled={false}
>
  <input
    className="c1"
    defaultChecked={false}
    disabled={false}
    id="random_check_box"
    onChange={[MockFunction]}
    size={20}
    type="checkbox"
  />
  <svg
    className="c2"
    data-name="checked"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6,3h12c1.7,0,3,1.3,3,3v12c0,1.7-1.3,3-3,3H6c-1.7,0-3-1.3-3-3V6C3,4.3,4.3,3,6,3z M10,17
	l9-8.6L17.6,7L10,14.3l-3.6-3.5L5,12.2L10,17z"
    />
  </svg>
  <svg
    className="c2"
    data-name="empty"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6,5C5.4,5,5,5.4,5,6v12c0,0.6,0.4,1,1,1h12c0.6,0,1-0.4,1-1V6c0-0.6-0.4-1-1-1H6z M6,3h12
	c1.7,0,3,1.3,3,3v12c0,1.7-1.3,3-3,3H6c-1.7,0-3-1.3-3-3V6C3,4.3,4.3,3,6,3z"
    />
  </svg>
</div>
`;

exports[`Checkbox updates and fires callback when clicked (when enabled) 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: #687B8E;
}

.c0 svg[data-name='checked'] {
  display: none;
}

.c0 > input:checked ~ svg[data-name='checked'] {
  display: inline-block;
  color: #007aff;
}

.c0 > input:checked ~ svg[data-name='empty'] {
  display: none;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
}

<div
  className="c0"
  disabled={false}
>
  <input
    className="c1"
    defaultChecked={false}
    disabled={false}
    id="random_check_box"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "target": Object {
                "checked": true,
              },
            },
          ],
        ],
      }
    }
    size={20}
    type="checkbox"
  />
  <svg
    className="c2"
    data-name="checked"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6,3h12c1.7,0,3,1.3,3,3v12c0,1.7-1.3,3-3,3H6c-1.7,0-3-1.3-3-3V6C3,4.3,4.3,3,6,3z M10,17
	l9-8.6L17.6,7L10,14.3l-3.6-3.5L5,12.2L10,17z"
    />
  </svg>
  <svg
    className="c2"
    data-name="empty"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6,5C5.4,5,5,5.4,5,6v12c0,0.6,0.4,1,1,1h12c0.6,0,1-0.4,1-1V6c0-0.6-0.4-1-1-1H6z M6,3h12
	c1.7,0,3,1.3,3,3v12c0,1.7-1.3,3-3,3H6c-1.7,0-3-1.3-3-3V6C3,4.3,4.3,3,6,3z"
    />
  </svg>
</div>
`;
